sogs/routes/rooms.py

```diff
@@ -301,7 +301,7 @@ def serve_file(room, fileId, filename):
     resp = make_response(send_from_directory(abspath(curdir), room_file.path, as_attachment=True))
 
     resp.headers["Date"] = http_date(room_file.uploaded)
-    resp.headers["Expires"] = http_date(room_file.expiry)
-    resp.headers["Content-Type"] = 'application/octet-stream'
+    if room_file.expiry and room_file.expiry > 0:
```

Comment:
0 is a special value that means "expire now", which isn't something we should really worry about here.  I think, however, that `model.room.Room.get_file` should be taking care of not returning expired files that haven't hit the cleanup job yet, i.e. by adding a not expired check here:
```Python
if row:
    return File(row)
```
i.e.:
```Python
if row and row['expiry'] > time.time():
    return File(row)
```

Reply:
(It'd be even better if it was in the query, except then we'd potentially trigger the extra file_id_hacks query crap unnecessarily).